packages/nbdime/src/common/mergeview.ts

```diff
@@ -143,563 +341,404 @@ function createNbdimeMergeView(
-      other.scrollTo(other.state.scrollPosition.left, other.state.scrollPosition.top);
-      other.state.scrollTicking = false;
-      other.state.scrollSetBy = self;
+    let srcScroller = srcEditor.scrollDOM;
+    let destScroller = destEditor.scrollDOM;
+    srcScroller.addEventListener("scroll", (event) => {
+      window.requestAnimationFrame(function() {destScroller.scrollLeft = srcScroller.scrollLeft;})
     });
-    other.state.scrollTicking = true;
     return;
   }
 
-
-  protected updateMarks(editor: CodeMirror.Editor, diff: DiffRangePos[],
-                        markers: Marker[], type: DIFF_OP) {
-    let classes: DiffClasses;
-    if (this.classes === null) {
-      // Only store prefixes here, will be completed later
-      classes = copyObj(mergeClassPrefix);
-    } else {
-      classes = this.classes;
-    }
-
-    let self = this;
-    function markChunk(editor: CodeMirror.Editor, from: number, to: number,
-                       sources: ChunkSource[]) {
-      if (self.classes === null && sources.length > 0) {
-        // Complete merge class prefixes here
-        classes = copyObj(mergeClassPrefix);
-        // First, figure out 'action' state of chunk
-        let s: string = sources[0].action;
-        if (sources.length > 1) {
-          for (let si of sources.slice(1)) {
-            if (si.action !== s) {
-              s = 'mixed';
-              break;
-            }
+  private getDecorationKey(sources: ChunkSource[]): string {
+    let s: string = this.type;
+    let res: string = s;
+    if (this.type === 'merge') {
+      s = sources[0].action;
+      res = s + 'Merge'
+      if (sources.length > 1) {
+        for (let si of sources.slice(1)) {
+          if (si.action !== s) {
+            res = 'mixedMerge';
+            break;
           }
         }
-        for (let k of Object.keys(classes)) {
-          classes[k] += '-' + s;
-        }
       }
-      // Next, figure out conflict state
-      let conflict = false;
-      if (sources.length > 0) {
-        for (let s of sources) {
-          if (s.decision.conflict) {
-            conflict = true;
-            break;
-          }
+    }
+    return res;
+  }
+
+  private getConflictState(sources: ChunkSource[]): boolean {
+    let conflict = false;
+    if (sources.length > 0) {
+      for (let s of sources) {
+        if (s.decision.conflict) {
+          conflict = true;
+          break;
         }
       }
+    }
+    return conflict;
+  }
+
+private createGutterEffects (editor: EditorView, chunk: Chunk, pos: number, on: true, type: string) {
+  let effects: StateEffect<unknown>[] = [];
+
+  let gutterEffect = addGutterMarkerEffect.of({
+    pos: pos,
+    on: on,
+    type: type
+  })
+  effects.push(gutterEffect);
+
+  let line: number = offsetToPos(editor.state.doc, pos).line;
+  let mappingEffect = addLineChunkMappingEffect.of({
+    line: line,
+    chunk: chunk,
+    type: type
+  });
+  effects.push(mappingEffect);
```

Comment:
Nitpicking: this function can be shorten by removing all the local variables (except effects), for instance:
```ts
effects.push(addLineChunkMappingEffect.of({
  line: offsetToPos(editor.state.doc, pos).line,
  chunk: chunk,
  type: type
}));
```

Reply:
Thanks. This has been taken into account.